`rtx ls-remote nodejs@16`

Currently you can't do a prefix search with `rtx ls-remote`:

```
$ rtx ls-remote nodejs@16
[ERROR] [nodejs@16] plugin not installed
[ERROR] Run with RTX_DEBUG=1 for more information.
```

That should show all nodejs releases with a prefix of "16".

--- a/README.md
+++ b/README.md
@@ -1020,20 +1020,31 @@ list runtime versions available for install
 note that these versions are cached for commands like `rtx install nodejs@latest`
 however _this_ command will always clear that cache and fetch the latest remote versions
 
-Usage: ls-remote <PLUGIN>
+Usage: ls-remote <PLUGIN> [PREFIX]
 
 Arguments:
   <PLUGIN>
-          Plugin
+          plugin to get versions for
+
+  [PREFIX]
+          the version prefix to use when querying the latest version same as the first argument after the "@"
 
 Options:
   -h, --help
           Print help (see a summary with '-h')
 
 Examples:
-  $ rtx list-remote nodejs
+  $ rtx ls-remote nodejs
   18.0.0
   20.0.0
+
+  $ rtx ls-remote nodejs@18
+  18.0.0
+  18.1.0
+
+  $ rtx ls-remote nodejs 18
+  18.0.0
+  18.1.0
 
 ```
 ### `rtx plugins install`

--- a/completions/_rtx
+++ b/completions/_rtx
@@ -466,7 +466,8 @@ _arguments "${_arguments_options[@]}" \
 '*--verbose[Show installation output]' \
 '-h[Print help (see more with '\''--help'\'')]' \
 '--help[Print help (see more with '\''--help'\'')]' \
-':plugin -- Plugin:' \
+':plugin -- plugin to get versions for:' \
+'::prefix -- the version prefix to use when querying the latest version same as the first argument after the "@":' \
 && ret=0
 ;;
 (plugins)

--- a/completions/rtx.bash
+++ b/completions/rtx.bash
@@ -1857,7 +1857,7 @@ _rtx() {
             return 0
             ;;
         rtx__ls__remote)
-            opts="-j -v -h --log-level --jobs --verbose --help <PLUGIN>"
+            opts="-j -v -h --log-level --jobs --verbose --help <PLUGIN> [PREFIX]"
             if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                 COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                 return 0

--- a/src/cli/ls_remote.rs
+++ b/src/cli/ls_remote.rs
@@ -1,5 +1,6 @@
-use atty::Stream;
+use crate::cli::args::runtime::{RuntimeArg, RuntimeArgParser, RuntimeArgVersion};
 use color_eyre::eyre::Result;
+use console::style;
 use indoc::formatdoc;
 use once_cell::sync::Lazy;
 
@@ -7,7 +8,6 @@ use crate::cli::command::Command;
 use crate::config::Config;
 use crate::errors::Error::PluginNotInstalled;
 use crate::output::Output;
-use crate::ui::color::Color;
 
 /// list runtime versions available for install
 ///
@@ -16,19 +16,37 @@ use crate::ui::color::Color;
 #[derive(Debug, clap::Args)]
 #[clap(visible_alias = "list-remote", verbatim_doc_comment, after_long_help = AFTER_LONG_HELP.as_str(), alias = "list-all")]
 pub struct LsRemote {
-    /// Plugin
+    /// plugin to get versions for
+    #[clap(value_parser = RuntimeArgParser)]
+    plugin: RuntimeArg,
+
+    /// the version prefix to use when querying the latest version
+    /// same as the first argument after the "@"
     #[clap()]
-    plugin: String,
+    prefix: Option<String>,
 }
 
 impl Command for LsRemote {
     fn run(self, config: Config, out: &mut Output) -> Result<()> {
         let plugin = config
             .plugins
-            .get(&self.plugin)
-            .ok_or(PluginNotInstalled(self.plugin))?;
+            .get(&self.plugin.plugin)
+            .ok_or(PluginNotInstalled(self.plugin.plugin))?;
         plugin.clear_remote_version_cache()?;
-        let versions = plugin.list_remote_versions(&config.settings)?;
+
+        let prefix = match self.plugin.version {
+            RuntimeArgVersion::Version(v) => Some(v),
+            _ => self.prefix,
+        };
+
+        let versions = plugin.list_remote_versions(&config.settings)?.clone();
+        let versions = match prefix {
+            Some(prefix) => versions
+                .into_iter()
+                .filter(|v| v.starts_with(&prefix))
+                .collect(),
+            None => versions,
+        };
 
         for version in versions {
             rtxprintln!(out, "{}", version);
@@ -38,25 +56,37 @@ impl Command for LsRemote {
     }
 }
 
-static COLOR: Lazy<Color> = Lazy::new(|| Color::new(Stream::Stdout));
 static AFTER_LONG_HELP: Lazy<String> = Lazy::new(|| {
     formatdoc! {r#"
     {}
-      $ rtx list-remote nodejs
+      $ rtx ls-remote nodejs
       18.0.0
       20.0.0
-    "#, COLOR.header("Examples:")}
+
+      $ rtx ls-remote nodejs@18
+      18.0.0
+      18.1.0
+
+      $ rtx ls-remote nodejs 18
+      18.0.0
+      18.1.0
+    "#, style("Examples:").bold().underlined()}
 });
 
 #[cfg(test)]
 mod tests {
-    use crate::assert_cli;
-    use crate::cli::tests::ensure_plugin_installed;
+    use insta::assert_snapshot;
+
+    use crate::assert_cli_snapshot;
 
     #[test]
     fn test_list_remote() {
-        ensure_plugin_installed("nodejs");
-        let stdout = assert_cli!("list-remote", "nodejs");
-        assert!(stdout.contains("18.0.0"));
+        assert_cli_snapshot!("list-remote", "dummy");
+    }
+
+    #[test]
+    fn test_ls_remote_prefix() {
+        assert_cli_snapshot!("list-remote", "dummy", "1");
+        assert_cli_snapshot!("list-remote", "dummy@2");
     }
 }

--- a/src/cli/mod.rs
+++ b/src/cli/mod.rs
@@ -238,6 +238,16 @@ pub mod tests {
     }
 
     #[macro_export]
+    macro_rules! assert_cli_snapshot {
+        ($($args:expr),+) => {{
+            let args = &vec!["rtx".into(), $($args.into()),+];
+            let output = $crate::cli::tests::cli_run(args).unwrap().stdout.content;
+            let output = console::strip_ansi_codes(&output).to_string();
+            assert_snapshot!(output);
+        }};
+    }
+
+    #[macro_export]
     macro_rules! assert_cli_err {
         ($($args:expr),+) => {{
             let args = &vec!["rtx".into(), $($args.into()),+];

--- a/src/cli/plugins/ls_remote.rs
+++ b/src/cli/plugins/ls_remote.rs
@@ -56,11 +56,9 @@ Examples:
 #[cfg(test)]
 mod tests {
     use crate::assert_cli;
-    use crate::cli::tests::ensure_plugin_installed;
 
     #[test]
     fn test_plugin_list_remote() {
-        ensure_plugin_installed("nodejs");
         let stdout = assert_cli!("plugin", "ls-remote");
         assert!(stdout.contains("nodejs"));
     }

